runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java

```diff
@@ -67,7 +68,7 @@ public Token appendCheckpoints(CorfuRuntime rt, String author) {
 
                 CheckpointWriter<T> cpw = new CheckpointWriter(rt, streamId, author, (T) map);
                 ISerializer serializer =
-                        ((CorfuCompileProxy<Map>) map.getCorfuSMRProxy())
+                        ((CorfuCompileProxy<SMRMap<String, String>>) map.getCorfuSMRProxy())
```

Comment:
I don't understand this change, why is it needed?

Reply:
Does not compile otherwise. Don't quite know why the compiler was not happy. I think `Map<String, String>` should work as well.

Reply:
You can drop the "map" cast and just use it a wrapped object. Why not use:

> `ISerializer serializer = ((CorfuCompileProxy) map.getCorfuSMRProxy())
                                .getSerializer();`

Reply:
Done.